mm/src/code/z_skelanime.c

```diff
@@ -1018,28 +1020,31 @@ void AnimationContext_SetLoadFrame(PlayState* play, PlayerAnimationHeader* anima
     AnimationEntry* entry = AnimationContext_AddEntry(&play->animationCtx, ANIMATION_LINKANIMETION);
 
     if (entry != NULL) {
-        if (ResourceMgr_OTRSigCheck(animation) != 0)
-            animation = ResourceMgr_LoadAnimByName(animation);
-
-        PlayerAnimationHeader* playerAnimHeader = Lib_SegmentedToVirtual(animation);
-        Vec3s* ram = frameTable;
-
-        // osCreateMesgQueue(&entry->data.load.msgQueue, &entry->data.load.msg, 1);
-        //
-        // char animPath[2048];
-        //
-        // snprintf(animPath, sizeof(animPath), "misc/link_animetion/gPlayerAnimData_%06X",
-        //         (((uintptr_t)linkAnimHeader->segmentVoid - 0x07000000)));
-        //
-        // printf("Streaming %s, seg = %08X\n", animPath, linkAnimHeader->segment);
-
-        s16* animData = /* ResourceMgr_LoadPlayerAnimByName*/ (animation->segmentVoid);
-        // 2S2H [Port] sometimes a HESS can set a negative frame value from a negative playback speed. When converted to
-        // a signed value this will cause a crash due to copying way much data.
-        if (frame < 0) {
-            frame = 0;
+        if (GameInteractor_Should(VB_LOAD_PLAYER_ANIMATION_FRAME, true, animation, frame, limbCount, frameTable)) {
```

Comment:
This whole block is already surrounded in a condition check, you can just use that instead of adding another layer of condition. Just make sure you check the `entry != NULL` in your VB handler.
```suggestion
    if (GameInteractor_Should(VB_LOAD_PLAYER_ANIMATION_FRAME, entry != NULL, animation, frame, limbCount, frameTable)) {
```